core/src/main/java/org/apache/iceberg/hadoop/HadoopTables.java

```diff
@@ -102,7 +102,7 @@ public Table load(String location) {
    */
   private Pair<String, MetadataTableType> parseMetadataType(String location) {
     int hashIndex = location.lastIndexOf('#');
-    if (hashIndex != -1 & !location.endsWith("#")) {
+    if (hashIndex != -1 && !location.endsWith("#")) {
```

Comment:
Brilliant. If we're going to consider enabling any more required rules, can something be done to better lint usages of boolean `&&` vs the bit operation? I imagine there's a rule for disallow implicit bit conversion to boolean or the like? That could be handled in a separate PR / issue (I'd be happy to look into it if you don't know off the top of your head).

Reply:
These are just some of the rules that were included by default when we upgraded the errorprone version when we upgraded the palanrir baseline plugin. I haven't looked at the full list but check baseline.gradlendor ones we have turned off explicitly